client/azure.go

```diff
@@ -174,6 +183,18 @@ func newDNSZonesClient(config AzureConfig) (*dns.ZonesClient, error) {
 	return &client, nil
 }
 
+func newVnetPeeringClient(config AzureConfig) (*network.VirtualNetworkPeeringsClient, error) {
+	spt, err := newServicePrincipalToken(config, azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		return nil, microerror.Maskf(err, "creating service principal token")
```

Comment:
I wanted to follow the already present structure here. But I also experienced harder debugging with those annotations, usually the traces speak by itself.

Would it make sense to you @kopiczko to removes those annotations in a different PR ?

Reply:
It's actually @xh3b4sd request. For me those annotations make sense but for me either approach isn't a blocker. I think for errors that we don't create it makes sense to add additional information in case they are cryptic.

Reply:
Just sharing my opinion here.